src/main/java/at/hannibal2/skyhanni/features/dungeon/DungeonFinderFeatures.kt

```diff
@@ -3,187 +3,352 @@ package at.hannibal2.skyhanni.features.dungeon
+        "price",
+        "([0-9]{2,3}K|[0-9]{1,3}M|[0-9]+\\.[0-9]M|[0-9] ?MIL)"
+    )
+    private val carryPattern by patternGroup.pattern(
+        "carry",
+        "(CARRY|CARY|CARRIES|CARIES|COMP|TO CATA [0-9]{2})"
+    )
+    private val nonPugPattern by patternGroup.pattern(
+        "nonpug",
+        "(PERM|VC|DISCORD)"
+    )
+    private val memberPattern by patternGroup.pattern(
+        "member",
+        ".*§.(?<playerName>.*)§f: §e(?<className>.*)§b \\(§e(?<level>.*)§b\\)"
+    )
+    private val ineligiblePattern by patternGroup.pattern(
+        "ineligible",
+        "§c(Requires .*$|You don't meet the requirement!|Complete previous floor first!$)"
+    )
+    private val classLevelPattern by patternGroup.pattern(
+        "class.level",
+        " §.(?<playerName>.*)§f: §e(?<className>.*)§b \\(§e(?<level>.*)§b\\)"
+    )
+    private val notePattern by patternGroup.pattern(
+        "note",
+        "(§7§7Note: |§f[^§])"
+    )
+    private val floorTypePattern by patternGroup.pattern(
+        "floor.type",
+        "(The Catacombs).*|.*(MM Catacombs).*"
+    )
+    private val checkIfPartyPattern by patternGroup.pattern(
+        "check.if.party",
+        ".*('s Party)"
+    )
+    private val partyFinderTitlePattern by patternGroup.pattern(
+        "party.finder.title",
+        "(Party Finder)"
+    )
+    private val catacombsGatePattern by patternGroup.pattern(
+        "catacombs.gate",
+        "(Catacombs Gate)"
+    )
+    private val selectFloorPattern by patternGroup.pattern(
+        "select.floor",
+        "(Select Floor)"
+    )
+    private val entranceFloorPattern by patternGroup.pattern(
+        "entrance",
+        "(.*Entrance)"
+    )
+    private val floorPattern by patternGroup.pattern(
+        "floor",
+        "(Floor .*)"
+    )
+    private val anyFloorPattern by patternGroup.pattern(
+        "floor.any",
+        "(Any)"
+    )
+    private val masterModeFloorPattern by patternGroup.pattern(
+        "floor.mastermode",
+        "(MM )|(.*Master Mode Catacombs)"
+    )
+    private val dungeonFloorPattern by patternGroup.pattern(
+        "floor.dungeon",
+        "(Dungeon: .*)"
+    )
+    private val floorFloorPattern by patternGroup.pattern(
+        "floor.pattern",
+        "(Floor: .*)"
+    )
+    private val floorNumberPattern by patternGroup.pattern(
+        "floor.number",
+        ".* (?<floorNum>[IV\\d]+)"
+    )
+    private val getDungeonClassPattern by patternGroup.pattern(
+        "get.dungeon.class",
+        ".* (?<class>.*)"
+    )
+    private val detectDungeonClassPattern by patternGroup.pattern(
+        "detect.dungeon.class",
+        "(View and select a dungeon class.)"
+    )
 
+    //  Variables used
     private var selectedClass = ""
+    private var floorStackSize = mapOf<Int, String>()
+    private var highlightParty = mapOf<Int, LorenzColor>()
+    private var toolTipMap = mapOf<Int, List<String>>()
+    private var inInventory = false
```

Comment:
Maybe extract this into three inventory booleans, for the three different inventories we could be in

Reply:
not necessary atm